tests/ft/type_checking/valid/empty_types.jk

```diff
@@ -0,0 +1,17 @@
+type Empty;
+e = Empty;
+
+type Ok;
+type NotOk;
+
+o = Ok;
```

Comment:
I'm not sure if this is a good design. Is `o` the type `Ok` or an instance of `Ok` (I know it's the latter but that a bit confusing).

Why not just go with the usual `Ok()` rather than creating a special edge case that complexify the language.

What is the benefit of this outside of removing the need of `()`

Reply:
Oops... that's my bad for having the wrong destination branch, but this discussion is more suited on #348. I'll answer there